### PR TITLE
Initial (basic) Behavior Designer integration

### PR DESCRIPTION
**Pull request by SorraTheOrc**

### CODE REVIEW

File: Scripts/IsAlive.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        private IHealthManager healthManager;
+        private GameObject prevGameObject;
+
+        public override void OnStart()
+        {
+            GameObject currentGameObject = GetDefaultGameObject(targetGameObject.Value);
+            if (currentGameObject != prevGameObject)
+            {
+                if (currentGameObject != null)
```
Comment (YondernautsGames) [inline Scripts/IsAlive.cs:24]: Should set healthManager to null if this fails

File: Scripts/IsAlive.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                {
+                    healthManager = currentGameObject.GetComponent<IHealthManager>();
+                }
+            }
+        }
+
+        public override TaskStatus OnUpdate()
+        {
+            if (healthManager.isAlive)
```
Comment (YondernautsGames) [inline Scripts/IsAlive.cs]: Need a null check against healthManager. Should this mean defaults to success or failure?

File: Scripts/DamageSource.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        }
+
+        public Transform damageSourceTransform
+        {
+            get { return transform; }
+        }
+
+        public string description
+        {
+            get { return transform.name + "inflicts damage of " + damageAmount; }
```
Comment (YondernautsGames) [inline Scripts/DamageSource.cs]: Description should just be a descriptive name (eg character name). It's up to the logger to format it in place

**Review by YondernautsGames — CHANGES_REQUESTED** [review]

**Comment by SorraTheOrc** [comment]
Thanks @YondernautsGames, these are pretty trivial (and resolved) so I'll go ahead and merge without further review.

**Pull request merged**